hepcrawl/spiders/aps_spider.py

```diff
@@ -123,7 +123,7 @@ def _parse_jats(self, response):
         file_name = self._file_name_from_url(response.url)
         parser.attach_fulltext_document(file_name, response.url)
         record = parser.parse()
-        file_urls = [document['url'] for document in record.get('documents', [])]
+        file_urls = [{"url": document['url'], "headers": {'Accept': 'text/xml'}} for document in record.get('documents', [])]
         return ParsedItem(
```

Comment:
`file_urls` doesn't take URLs any longer, but a list of `dict`s now, so the name doesn't match. Maybe we should change the API to take a list of `Request`s instead?

Reply:
I don't understand. In results scrapy is producing list of tuples with url in there:
`2020-10-23 09:24:02 [hepcrawl.pipelines] INFO: results: [(True, {'url': 'http://aps-http-server.local/PhysRevD.96.095036.xml', 'path': 'full/b99616c5061a542667fb4fa1d5a8ab750a15c731.xml', 'checksum': '4d8a21a5aeb643400328ed1915e70f66'})]`
And then we are matching this url with url in record. Looks lie it still works. there are even tests to check if document url in record is what we want. And it's still passing.

Reply:
I'm not saying it's broken, I'm saying you're making it more complex by:
* having content of `file_urls` not match the name (not a list of URLs any longer)
* needing to pass a magic dict, which turns out to be `kwargs` to `Request`

My suggestion:
* pass a list of `Request`s to `ParsedItem` instead of only its args
* replace the `ParsedItem` `file_urls` parameter by something more explicit, like `document_requests`